crates/optimism/primitives/src/transaction/signed.rs

```diff
@@ -228,7 +229,7 @@ impl alloy_rlp::Encodable for OpTransactionSigned {
 
     fn length(&self) -> usize {
         let mut payload_length = self.encode_2718_len();
-        if !self.is_legacy() {
+        if !Encodable2718::is_legacy(self) {
```

Comment:
In some parts, we have to specify `Encodable2718` or `Typed2718` specifically, but the result will be the same.
Should we consolidate the `is_legacy` method to one or the other?

Reply:
right, I think Encodable2718 should just have Typed2718 as supertrait